### PR TITLE
MET-44: Change frontend components to use the new measurement system

### PR DESCRIPTION
**Pull request by ValentinMumble**
**Description (for Contributor and Core Developer)**

- This PRs modifies the frontend views to use the new Measurement system labels instead of the Crowdin translations.

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | OK
| Added legacy Behats               | -
| Added acceptance tests            | -
| Added integration tests           | OK
| Changelog updated                 | OK
| Review and 2 GTM                  | Ok
| Micro Demo to the PO (Story only) | -
| Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

File: src/Akeneo/Platform/Bundle/UIBundle/Resources/public/js/channel/form/properties/conversion-unit.js [hunk 1]
```diff
@@ -58,17 +58,16 @@ define([
                     }),
                     FetcherRegistry.getFetcher('measure').fetchAll()
                 ).then(function (attributes, measures) {
-
                     this.$el.html(this.template({
                         conversionUnits: this.getFormData().conversion_units,
                         metrics: attributes,
-                        measures: measures,
```
Comment (SamirBoulil) [inline src/Akeneo/Platform/Bundle/UIBundle/Resources/public/js/channel/form/properties/conversion-unit.js]: Is it normal we inject measures like this ? ( I need to check in a proper screen)

Comment (ValentinMumble) [reply src/Akeneo/Platform/Bundle/UIBundle/Resources/public/js/channel/form/properties/conversion-unit.js]: Well I tried to edit the less I could, the new measures here are handled by the template:
https://github.com/akeneo/pim-community-dev/pull/11630/files#diff-55d516b1c8fb07f9948edf76971445a3R17

File: src/Akeneo/Tool/Bundle/MeasureBundle/Model/MeasurementFamily.php [hunk 2]
```diff
@@ -86,10 +86,14 @@ public function getUnitLabel(UnitCode $unitCode, LocaleIdentifier $localeIdentif
     {
         $unit = $this->getUnit($unitCode, $this->units);
 
+        if (!$unit) {
```
Comment (SamirBoulil) [inline src/Akeneo/Tool/Bundle/MeasureBundle/Model/MeasurementFamily.php]: Suggestion
```suggestion
        if (null === $unit) {
```

File: src/Akeneo/Pim/Enrichment/Component/Product/Localization/Presenter/MetricPresenter.php [hunk 3]
```diff
@@ -42,14 +46,18 @@ public function present($value, array $options = [])
             $value = $this->getStructuredMetric($value, $options['versioned_attribute']);
         }
 
+        $measurementFamilyCode = $this->baseCachedObjectRepository->findOneByIdentifier($options['attribute_code'])
```
Comment (tseho) [inline src/Akeneo/Pim/Enrichment/Component/Product/Localization/Presenter/MetricPresenter.php]: Where do you validate if `$options` contains `attribute_code` ?

File: src/Akeneo/Tool/Bundle/MeasureBundle/Model/MeasurementFamily.php [hunk 4]
```diff
@@ -87,4 +81,31 @@ function (Unit $unit) {
...
+    private function getUnit(UnitCode $standardUnitCode, array $units): ?Unit
+    {
+        $unit = current(array_filter(
+            $units,
+            function (Unit $unit) use ($standardUnitCode) {
+                return $standardUnitCode->equals($unit->code());
+            }
+        ));
+
+        if (!$unit) {
```
Comment (tseho) [inline src/Akeneo/Tool/Bundle/MeasureBundle/Model/MeasurementFamily.php]:
```suggestion
        if (null === $unit) {
```

Comment (ValentinMumble) [reply src/Akeneo/Tool/Bundle/MeasureBundle/Model/MeasurementFamily.php]: `current` returns false if the array is empty ;)

File: src/Akeneo/Pim/Enrichment/Component/Product/Localization/Presenter/MetricPresenter.php (hunk 3 above)
Comment (SamirBoulil) [reply src/Akeneo/Pim/Enrichment/Component/Product/Localization/Presenter/MetricPresenter.php]: nowhere, that's true :/ we should add the check at the beginning of the function

**Pull request merged**